### PR TITLE
VZ 8965: BFS Kontainer-engine-driver-ociocne

### PR DESCRIPTION
**Pull request by anders-swanson**
VZ 8965: Include basic kontainer driver for OCI OCNE Driver BFS

### CODE REVIEW

**Comment by kminder** [comment]
Needs all of the "standard" repo files.  Possibly non-exhaustive list: LICENSE.txt, SECURITY.md, THIRD_PARTY_LICENSES.txt, NOTICE.txt, CONTRIBUTING.md etc.

**Comment by kminder** [comment]
Which files are derivative works (if any)?

**Comment by anders-swanson** [comment]
> Which files are derivative works (if any)?

CSI, CCM YAMLs are based off oracle CSI, CCM release objects. All other files are original

File: NOTICE.txt
Comment (kminder) [inline NOTICE.txt:1]: Does this need to have attributions for repos from which things were copied?

File: THIRD_PARTY_LICENSES.txt
Comment (kminder) [inline THIRD_PARTY_LICENSES.txt:1]: Did this come from the BA?

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+replace (
+	k8s.io/api v0.26.0 => k8s.io/api v0.25.4
+	k8s.io/apimachinery v0.26.0 => k8s.io/apimachinery v0.25.4
+	k8s.io/client-go v0.26.0 => k8s.io/client-go v0.25.4
+)
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
```
Comment (kminder) [inline go.mod:26]: Do we have PLS approval for everything in here?

File: ociocne/capi/capi.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	{gvr.Cluster, templates.Cluster},
+	{gvr.OCICluster, templates.OCICluster},
+	{gvr.KubeadmConfigTemplate, templates.OCNEConfigTemplate},
+	{gvr.KubeadmControlPlane, templates.OCNEControlPlane},
+	{gvr.MachineDeployment, templates.MachineDeployment},
+	{gvr.OCIMachineTemplate, templates.OCIMachineTemplate},
+	{gvr.OCIMachineTemplate, templates.OCIControlPlaneMachineTemplate},
+}
+
+func CreateOrUpdateAllObjects(ctx context.Context, client dynamic.Interface, v *variables.Variables) error {
```
Comment (kminder) [inline ociocne/capi/capi.go:51]: Exported functions should have comments.

File: ociocne/templates/ccmconfigmap.goyaml
Comment (kminder) [inline ociocne/templates/ccmconfigmap.goyaml:1]: Is this one of the files that was derived from elsewhere?
Where they originally copyrighted?  If so we can't add Oracle copyright?
If this is a derivative file I don't think it should include the license.
If this is a derivative file we should add attribution information to the file.

File: go.mod (hunk 1 above)
Comment (anders-swanson) [reply go.mod:26]: We don't do this for indirect dependencies. For other dependencies, we are good

File: THIRD_PARTY_LICENSES.txt
Comment (anders-swanson) [reply THIRD_PARTY_LICENSES.txt:1]: I generated this, should I get it from somewhere else?

File: NOTICE.txt
Comment (anders-swanson) [reply NOTICE.txt:1]: For calico, yes.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+.idea/
+dist/
+integ_test.go
```
Comment (kminder) [inline .gitignore:6]: Why is integ_test.go in here?

File: NOTICE.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+The Initial Developer of ociocne/templates/calicoconfigmap.goyaml,
```
Comment (kminder) [inline NOTICE.txt:4]: This is fine for now if this is the only file involved.  Over time if more files are involved it is probably better if derivative files are isolated by directory or package then the notice could be more generic.  For example The initial developer of the files in ... is ...

Comment (anders-swanson) [reply NOTICE.txt:4]: OK, we will change the notice format if additional files are added

File: .gitignore (hunk 3 above)
Comment (anders-swanson) [reply .gitignore:6]: Yes, it will be resolved soon as I add testing

**Comment by anders-swanson** [comment]
> What is the related ticket? Needs PR description.

Added ticket number

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**